doc/api/n-api.md

```diff
@@ -2046,6 +2046,49 @@ the `napi_value` in question is of the JavaScript type expected by the API.
 
 ### Enum types
 
+#### `node_api_features`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+```c
+typedef enum {
+  node_api_feature_none = 0,
+  node_api_feature_reference_all_types = 1 << 0,
+
+  node_api_default_experimental_features = node_api_feature_reference_all_types,
+
+   // version specific
+  node_api_default_features = node_api_default_experimental_features,
```

Comment:
For compatiblity I'm wondering if we should tie the defaults to a Node-API version versus the vesion of Node.js Unless the module author choses to opt-in, they behaviour should not change unless they have moved up to a newer Node-API vesion?

Reply:
Looking more closely I think that is the approach that is described, but but nothing is guarded because its under experimental.  Will make a comment in src/js_native_api_types.h

Reply:
My intent is to make it per Node-API version.